File: Beginner/Ambiguous Permutations/Amb_Permu.js
const readline = require("prompt-sync")({sigint: true});
let n = Number(readline());
while (n!=0) {
    let num = new Array(n), zeros = new Array(n);
    let arr = readline().split(" ");
    for (let i=1; i<=n; i++) {
        num[i-1] = i;
        zeros[i-1] = 0;
    }
    for (let j=0; j<n; j++) {
        zeros[arr[j]-1] = num[j];
    }
    let check = new Boolean(true);
    for (let k=0; k<n; k++) {
        if (arr[k] !== zeros[k]) {
            check=false;
        }
    }
    if (check) {
        console.log("ambiguous");
    } else {
        console.log("not ambiguous");
    }
    n = Number(readline());
}

File: Beginner/Ambiguous Permutations/Amb_Permu.py
while True:
    arr_sz = int(input())
    if arr_sz == 0:
        break
    arr = list(map(int, input().split()))
    num = [i for i in range(1, arr_sz+1)]
    zeros = [0]*arr_sz
    for i in range(arr_sz):
        zeros[arr[i]-1] = num[i]
    if arr == zeros:
        print("ambiguous")
    else:
        print("not ambiguous")
